### PR TITLE
Fixes for issues found when adding ODD integration tests to .NET SDK

### PR DESCRIPTION
**Pull request by XDex**
## Description

Fixes for issues found when adding ODD integration tests to .NET SDK
Tests are failing at this point, will need to fix those in subsequent commits..

### CODE REVIEW

**Comment by XDex** [comment]
@artur-ciocanu I've reverted `onDeviceAllMatchingRulesMboxes` removal, please take another look, thx!

File: src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningDetailsExecutor.java [hunk 1]
```diff
@@ -312,12 +318,12 @@ private String requestPropertyToken(TargetDeliveryRequest deliveryRequest) {
   }
 
   private boolean propertyTokenMismatch(List<String> rulePropertyTokens, String propertyToken) {
-    if (StringUtils.isEmpty(propertyToken)) {
-      return false;
-    }
     if (rulePropertyTokens == null || rulePropertyTokens.isEmpty()) {
       return false;
     }
```
Comment (artur-ciocanu) [inline src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningDetailsExecutor.java:323]: Why we have reordered the conditions?

**Review by artur-ciocanu — APPROVED** [review]
@XDex looks good I have left a few comments. i am not sure why you have removed all the property tokens from JSON file. We need some tests around property tokens unless those are covered in other places.

File: src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningDetailsExecutor.java (hunk 1 above)
Comment (XDex) [reply src/main/java/com/adobe/target/edge/client/ondevice/OnDeviceDecisioningDetailsExecutor.java:323]: We need to check if a rule doesn't have any property tokens _before_ checking if there's no property supplied in the request, as if there's no property in the request we still need to get offers from rules that don't have a property token.

**Pull request merged**